io/plugins/eu.esdihumboldt.hale.io.haleconnect/src/eu/esdihumboldt/hale/io/haleconnect/internal/HaleConnectServiceImpl.java

```diff
@@ -156,11 +174,197 @@ public void removeListener(HaleConnectServiceListener listener) {
+								int statusCode, Map<String, List<String>> responseHeaders) {
+							future.setException(new HaleConnectException(e.getMessage(), e,
+									statusCode, responseHeaders));
+						}
+
+						@Override
+						public void onSuccess(List<BucketDetail> result, int statusCode,
+								Map<String, List<String>> responseHeaders) {
+							future.set(Collections.unmodifiableList(processBucketDetails(result)));
+						}
+
+						@Override
+						public void onUploadProgress(long bytesWritten, long contentLength,
+								boolean done) {
+							// Ignored
+						}
+
+						@Override
+						public void onDownloadProgress(long bytesRead, long contentLength,
+								boolean done) {
+							// Ignored
+						}
+					});
+		} catch (com.haleconnect.api.projectstore.v1.ApiException e) {
+			throw new HaleConnectException(e.getMessage(), e);
+		}
+
+		return future;
+	}
+
+	/**
+	 * Convert a list of {@link BucketDetail}s received from the project store
+	 * to a list of {@link HaleConnectProjectInfo}
+	 * 
+	 * @param bucketDetails bucket details
+	 * @return list of hale connect project info
+	 */
+	private List<HaleConnectProjectInfo> processBucketDetails(List<BucketDetail> bucketDetails) {
+		List<HaleConnectProjectInfo> result = new ArrayList<>();
+		for (BucketDetail bucket : bucketDetails) {
+			if (bucket.getId() != null) {
+				String author = null;
+				if (bucket.getProperties() instanceof Map<?, ?>) {
+					@SuppressWarnings("unchecked")
+					Map<Object, Object> properties = (Map<Object, Object>) bucket.getProperties();
+					if (properties.containsKey("author")) {
+						author = properties.get("author").toString();
+					}
+				}
+
+				HaleConnectUserInfo user = null;
+				HaleConnectOrganisationInfo org = null;
+				try {
+					if (!StringUtils.isEmpty(bucket.getId().getUserId())) {
+						user = this.getUserInfo(bucket.getId().getUserId());
+					}
+
+					if (!StringUtils.isEmpty(bucket.getId().getOrgId())) {
+						org = this.getOrganisationInfo(bucket.getId().getOrgId());
+					}
+				} catch (HaleConnectException e) {
+					log.error(e.getMessage(), e);
+				}
+
+				result.add(new HaleConnectProjectInfo(bucket.getId().getTransformationproject(),
+						user, org, bucket.getName(), author));
+			}
+		}
+		return result;
+	}
+
+	/**
+	 * @see eu.esdihumboldt.hale.io.haleconnect.HaleConnectService#loadProject(Owner,
+	 *      String)
+	 */
+	@Override
+	public LocatableInputSupplier<InputStream> loadProject(Owner owner, String projectId) {
+		return new HaleConnectInputSupplier(projectId, owner,
+				this.getBasePath(HaleConnectServices.PROJECT_STORE), this.getSession().getToken());
+	}
+
+	private String getContextOrganisation() {
+		if (!this.isLoggedIn()) {
+			return null;
+		}
+
+		List<String> orgIds = this.getSession().getOrganisationIds();
+		if (orgIds.isEmpty()) {
+			return null;
+		}
+
+		// XXX Cannot handle multiple organisations!
+		return orgIds.iterator().next();
```

Comment:
My impression was that multiple organisation roles are not supported by hale connect at the moment. If that changes, the user would need a way to choose an "active" organisation in hale connect.